orgs/organization.go

```diff
@@ -0,0 +1,144 @@
+		return nil
+	}
+	token, err := o.auth0Client.ValidateToken(o.Environment)
+	if err != nil {
+		log.Println(err)
+	}
+	headers := make(map[string]string, 0)
+	headers["Content-type"] = "application/json"
+	headers["Authorization"] = fmt.Sprintf("%s %s", "Bearer", token)
+
+	if offset == "" {
+		offset = "0"
+	}
+	if pageSize == "" {
+		pageSize = "100"
+	}
+	endpoint := o.OrgBaseURL + "/orgs/search?name=" + name + "&pageSize=" + pageSize + "&offset=" + offset
+	_, res, err := o.httpClient.Request(strings.TrimSpace(endpoint), "GET", headers, nil, nil)
+	if err != nil {
+		log.Println("SearchOrganization: Could not get the organization list: ", err)
+		return nil
+	}
+	var response SearchOrganizationResponse
+	err = json.Unmarshal(res, &response)
+	if err != nil {
+		log.Println("SearchOrganization: failed to unmarshal SearchOrganizationResponse: ", err)
+		return nil
+	}
+	return &response
+}
+
+// LookupOrganization ...
+func (o *Org) LookupOrganization(name string) *Organization {
+	if name == "" {
+		log.Println("LookupOrganization: name is empty")
+		return nil
+	}
+	token, err := o.auth0Client.ValidateToken(o.Environment)
+	if err != nil {
+		log.Println(err)
+	}
+	headers := make(map[string]string, 0)
+	headers["Content-type"] = "application/json"
+	headers["Authorization"] = fmt.Sprintf("%s %s", "Bearer", token)
+
+	endpoint := o.OrgBaseURL + "/lookup?name=" + name
+	_, res, err := o.httpClient.Request(strings.TrimSpace(endpoint), "GET", headers, nil, nil)
+	if err != nil {
+		log.Println("LookupOrganization: Could not get the organization: ", err)
+		return nil
+	}
+	var response Organization
+	err = json.Unmarshal(res, &response)
+	if err != nil {
+		log.Println("LookupOrganization: failed to unmarshal LookupOrganizationResponse: ", err)
+		return nil
+	}
+	return &response
+}
+
+// NewOrganizationsClient consumes
+// orgBaseURL, esCacheUrl, esCacheUsername, esCachePassword, esCacheIndex, env, authGrantType, authClientID, authClientSecret, authAudience, authURL
+func NewOrganizationsClient(orgBaseURL, esCacheURL, esCacheUsername,
+	esCachePassword, env, authGrantType, authClientID, authClientSecret,
+	authAudience, authURL string) (*Org, error) {
+	org := &Org{
+		OrgBaseURL:       orgBaseURL,
+		ESCacheURL:       esCacheURL,
+		ESCacheUsername:  esCacheUsername,
+		ESCachePassword:  esCachePassword,
+		AuthGrantType:    authGrantType,
+		AuthClientID:     authClientID,
+		AuthClientSecret: authClientSecret,
+		AuthAudience:     authAudience,
+		AuthURL:          authURL,
+		Environment:      env,
+	}
+
+	httpClientProvider, esClientProvider, auth0ClientProvider, err := buildServices(org)
+	if err != nil {
+		return nil, err
+	}
+
+	org.esClient = esClientProvider
+	org.httpClient = httpClientProvider
+	org.auth0Client = auth0ClientProvider
+
+	return org, nil
+}
+
+func buildServices(o *Org) (httpClientProvider *http.ClientProvider, esClientProvider *elastic.ClientProvider, auth0ClientProvider *auth0.ClientProvider, err error) {
```

Comment:
Based on Golang codeReview doc, named output params is not good approach, remove the names and return explicit vars

Reply:
done!